service/discovery/azure/storage.go

```diff
@@ -256,69 +325,98 @@ func (d *azureStorageDiscovery) discoverSqlServers() ([]voc.IsCloudResource, err
-		dbService voc.IsCloudResource
-		list      []voc.IsCloudResource
-		err       error
+		dbList               []voc.IsCloudResource
+		anomalyDetectionList []voc.IsAnomalyDetection
+		dbService            voc.IsCloudResource
+		list                 []voc.IsCloudResource
+	)
+
+	// Get SQL database storages and the corresponding anomaly detection property
+	dbList, anomalyDetectionList = d.getSqlDBs(server)
+
+	// Create SQL database service voc object for SQL server
+	// TODO(anatheka): Delete?
+	// TODO(oxisto): This is not 100 % accurate. According to our ontology definition, the SQL server would be
+	// the database service and individual databases would be DatabaseStorage objects. However, the problem is
+	// that azure defines anomaly detection on a per-database level and we currently have anomaly detection as
+	// part of the service
+	dbService = &voc.DatabaseService{
+		StorageService: &voc.StorageService{
+			NetworkService: &voc.NetworkService{
+				Networking: &voc.Networking{
+					Resource: discovery.NewResource(d,
+						voc.ResourceID(*server.ID),
+						*server.Name,
+						nil, // creation time not available
+						voc.GeoLocation{
+							Region: *server.Location,
+						},
+						labels(server.Tags),
+						resourceGroupID(server.ID),
+						voc.DatabaseServiceType,
+						server,
+					),
+				},
+				// TODO(all): HttpEndpoint
+				TransportEncryption: &voc.TransportEncryption{
+					Enabled:    true,
+					Enforced:   true,
+					TlsVersion: *server.Properties.MinimalTLSVersion,
+				},
+			},
+		},
+		AnomalyDetection: anomalyDetectionList,
+	}
+
+	// Add SQL database service
+	list = append(list, dbService)
+
+	// Add SQL database storages
+	list = append(list, dbList...)
+
+	return list, nil
+}
+
+// getSqlDBs returns a list of SQL databases for a specific SQL account
+func (d *azureStorageDiscovery) getSqlDBs(server *armsql.Server) ([]voc.IsCloudResource, []voc.IsAnomalyDetection) {
+	var (
+		list                 []voc.IsCloudResource
+		anomalyDetectionList []voc.IsAnomalyDetection
+		err                  error
 	)
 
 	// initialize SQL databases client
 	if err = d.initDatabasesClient(); err != nil {
-		return nil, err
+		log.Errorf("error initializing database client: %v", err)
+		return list, anomalyDetectionList
 	}
 
 	// Get databases for given server
 	serverlistPager := d.clients.databasesClient.NewListByServerPager(resourceGroupName(util.Deref(server.ID)), *server.Name, &armsql.DatabasesClientListByServerOptions{})
 	for serverlistPager.More() {
 		pageResponse, err := serverlistPager.NextPage(context.TODO())
 		if err != nil {
-			err = fmt.Errorf("%s: %v", ErrGettingNextPage, err)
-			return nil, err
+			log.Errorf("%s: %v", ErrGettingNextPage, err)
+			return list, anomalyDetectionList
 		}
 
 		for _, value := range pageResponse.Value {
-			// Getting anomaly detection status
-			anomalyDetectionEnabeld, err := d.anomalyDetectionEnabled(server, value)
+			// Create anomaly detection property
```

Comment:
@oxisto @lebogg regarding the TODO of oxisto do we want the anomalyDetection as part of the databaseService or databaseStorage? I have done it as a list of anomalyDetections in the databaseService, but I'm not sure if that is a good idea. What do you think?
 
```
// TODO(oxisto): This is not 100 % accurate. According to our ontology definition, the SQL server would be
// the database service and individual databases would be DatabaseStorage objects. However, the problem is
// that azure defines anomaly detection on a per-database level and we currently have anomaly detection as
// part of the service
```

Reply:
Sounds interesting, the only problem that I see is that we cannot differentiate between the different anomaly detection objects, can we?

Reply:
I've added an additional field 'resource' for the database resource ID, that could be a possibility. 
![grafik](https://github.com/clouditor/clouditor/assets/18168509/b5aecb2d-6bd9-4dff-a109-a05e5e5181e1)

Reply:
We changed it to `Scope` to avoid confusion.